File: Tema2/app/src/main/java/com/example/tema2/activities/MainActivity.java
package com.example.tema2.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;

import com.example.tema2.R;
import com.example.tema2.adapters.StatePagerAdapter;
import com.example.tema2.fragments.AlbumsFragment;
import com.example.tema2.fragments.PhotosFragment;
import com.example.tema2.fragments.UsersFragment;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "SecondActivity";

    private StatePagerAdapter statePagerAdapter;
    private ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        statePagerAdapter = new StatePagerAdapter(getSupportFragmentManager());

        viewPager = (ViewPager) findViewById(R.id.fragment_container);

        setupViewPager(viewPager);
    }

    private void setupViewPager(ViewPager viewpager){
        StatePagerAdapter adapter = new StatePagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new UsersFragment(), "UsersFragment");
        adapter.addFragment(new AlbumsFragment(), "AlbumsFragment");
        adapter.addFragment(new PhotosFragment(), "PhotosFragment");

        viewpager.setAdapter(adapter);
    }

    public void setViewPager(int fragmentNumber){
        viewPager.setCurrentItem(fragmentNumber);
    }
}

File: Tema2/app/src/main/java/com/example/tema2/activities/MainActivity.kt
package com.example.tema2.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.tema2.R
import com.example.tema2.fragments.UsersFragment
import com.example.tema2.fragments.AlbumsFragment
import com.example.tema2.interfaces.FragmentCommunication
import com.example.tema2.models.User

class MainActivity : AppCompatActivity(), FragmentCommunication {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        addUsersFragment()
    }

    private fun addUsersFragment() {
        val fragmentManager = supportFragmentManager
        val transaction = fragmentManager.beginTransaction()
        val tag = UsersFragment::class.java.name
        val addTransaction = transaction.add(
                R.id.frame_layout, UsersFragment.newInstance("", ""), tag
        )
        addTransaction.commit()
    }

    override fun addAlbumsFragment(user: User) {
        val fragmentManager = supportFragmentManager
        val transaction = fragmentManager.beginTransaction()
        val tag = AlbumsFragment::class.java.name
        val replaceTransaction = transaction.replace(
                R.id.frame_layout, AlbumsFragment.newInstance("", ""), tag
        )
        replaceTransaction.addToBackStack(tag)
        replaceTransaction.commit()
    }
}